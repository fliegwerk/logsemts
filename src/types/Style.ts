export interface StyleInterface {
	color: string;
	background: string;
	readonly css: string;
}

/**
 *
 */
export default class Style implements StyleInterface {
	constructor(public color: string, public background: string) {}

	get css() {
		return `color: ${this.color}; background-color: ${this.background}; font-weight: bold;`;
	}

	toJSON() {
		return { color: this.color, background: this.background, css: this.css };
	}
}
