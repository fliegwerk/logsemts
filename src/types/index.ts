export * from './LogFunction'
export * from './LogFunctionFactory'
export * from './Options'
export * from './Style'
