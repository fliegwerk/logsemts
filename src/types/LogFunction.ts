import Style from './Style';

type LogFunction = (type: string, style: Style, ...args: any[]) => void;

export default LogFunction;
