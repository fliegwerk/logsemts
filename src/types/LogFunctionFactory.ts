import { LogFunction } from './LogFunction';

/**
 * Builds a {@link LogFunction}
 * @param options for the log function
 * @return the {@link LogFunction}
 */
export type LogFunctionFactory = (...args: any[]) => LogFunction;
