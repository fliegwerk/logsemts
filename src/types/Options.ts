import LogFunction from './LogFunction';

/**
 * Options for a logger
 *
 * @see {@link Logger}
 */
export interface Options {
	/**
	 * The `LogFunction`s to which log messages get forwarded.
	 *
	 * @see {@link LogFunction}
	 * @see {@link Logger}
	 */
	loggers: LogFunction[];
}
