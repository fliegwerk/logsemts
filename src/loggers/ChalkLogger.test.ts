import ChalkLogger from './ChalkLogger';
import { StyleInterface } from '../types/Style';

import ch from 'chalk';
jest.mock('chalk');
import { mocked } from 'ts-jest/utils';
import Mock = jest.Mock;

const origLog = console.log;

const style1: StyleInterface = {
	background: 'red',
	css: '',
	color: '#fff'
};

const style2: StyleInterface = {
	color: 'green',
	background: '#ffffff',
	css: ''
};

const invalidStyle = {
	color: 'thiscolordoesnotexist',
	background: 'thiscolordoesnotexist',
	css: ''
};

describe('ChalkLogger', () => {
	let mockLog: Mock<any, any>;

	beforeEach(() => {
		mockLog = jest.fn();
		console.log = mockLog;
	});

	afterEach(() => {
		console.log = origLog;
	});

	it('should log valid inputs correctly', () => {
		ChalkLogger(mocked(ch))(
			'TEST',
			style1,
			'ChalkLogger Test',
			style2,
			'Hello world'
		);

		expect(mockLog).toHaveBeenCalled();
		expect(mockLog).toMatchSnapshot();
	});

	it('should ignore invalid color names', () => {
		ChalkLogger(mocked(ch))(
			'Test',
			invalidStyle,
			'ChalkLogger Test',
			invalidStyle,
			'Hello world'
		);
		expect(mockLog).toMatchSnapshot();
	});
});
