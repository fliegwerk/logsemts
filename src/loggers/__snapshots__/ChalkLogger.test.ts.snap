// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChalkLogger should ignore invalid color names 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[bold]( Test )[bold]( ChalkLogger Test )",
      "Hello world",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`ChalkLogger should log valid inputs correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[bold]([hex,#fff]([bgHex,#ff0000]( TEST )))[bold]([hex,#008000]([bgHex,#ffffff]( ChalkLogger Test )))",
      "Hello world",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
